### PR TITLE
feat: Introduce Analyzer to check for Unhandled exceptions, also in called methods

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
Extension of PH2120, which also checks called methods for unhandled exceptions. Relies on other methods documenting their unhandled exceptions.

Added a few mehtods form the .NET runtime. To have more coverage for runtime methods we need another mechanism.

### CODE REVIEW

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentUnhandledExceptionsAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+			public const string IoException = "System.IO.IOException";
+			public const string NotSupportedException = "System.NotSupportedException";
+			public const string PathTooLongException = "System.IO.PathTooLongException";
+			public const string SecurityException = "System.SecurityException";
+			public const string UnauthorizedException = "System.UnauthorizedException";
+		}
+
+		private static readonly DiagnosticDescriptor Rule =
+			new(Helper.ToDiagnosticId(DiagnosticIds.DocumentUnhandledExceptions), Title, MessageFormat, Category,
+				DiagnosticSeverity.Error, isEnabledByDefault: false, description: Description);
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentUnhandledExceptionsAnalyzer.cs:44]: Why disabled by default?

File: Philips.CodeAnalysis.Common/DiagnosticIds.cs [hunk 2]
```diff
@@ -116,5 +116,6 @@ public enum DiagnosticIds
 		ThrowInformationalExceptions = 2121,
 		AvoidExceptionsFromUnexpectedLocations = 2122,
 		PassSenderToEventHandler = 2123,
+		DocumentUnhandledExceptions = 2124,
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: PH2121 fired on this:
throw new ArgumentException($"The paths '{fromDirectory} and '{toPath}' have different path roots.");   [Setting "Requested Changes" for this one. Other feedback is at your judgement.]

Comment (bcollamore) [inline Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: PH2020: read "... from an NotImplementedException". Try different language to avoid "a" vs "an".

Comment (bcollamore) [inline Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: PH2122: Ideally the squiggle would be with the throw rather than the method name.

**Review by bcollamore — CHANGES_REQUESTED** [review]
CodeFixers! These will make it much more likely that people will adopt these on legacy code bases.

File: Philips.CodeAnalysis.Common/DiagnosticIds.cs (hunk 2 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: Nice catch. Added this as a test case and fixed the Analyzer

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentUnhandledExceptionsAnalyzer.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/DocumentUnhandledExceptionsAnalyzer.cs:44]: Oops, this is an error

File: Philips.CodeAnalysis.Common/DiagnosticIds.cs (hunk 2 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: Rephrased to avoid the "a" vs "an" issue

Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Common/DiagnosticIds.cs:119]: I agree, fixed

**Review by bcollamore — APPROVED** [review]

**Pull request merged**